pkg/cmd/cli/cmd/env.go

```diff
@@ -285,12 +302,25 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 		if err != nil {
 			return err
 		}
-		return p.PrintObj(objects, out)
+		for _, object := range objects {
+			if err := p.PrintObj(object, out); err != nil {
+				return err
+			}
+		}
+		return nil
 	}
 
 	failed := false
-	for _, info := range infos {
-		obj, err := resource.NewHelper(info.Client, info.Mapping).Replace(info.Namespace, info.Name, true, info.Object)
+	for i, info := range infos {
+		newData, err := json.Marshal(objects[i])
+		if err != nil {
+			return err
+		}
+		patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData[i], newData, objects[i])
```

Comment:
do we have good tests for these cases, to make sure patching is correctly handling:
* adding env var to a nil list
* adding env var to an empty list
* adding env var to a list of existing env vars
* updating single env var
* updating multiple env vars (at the beginning, middle, and end of a list)
* removing env var from a list containing only that env var
* removing env var from the beginning, middle, and end of a list containing other envvars